packages/node_modules/@ciscospark/spark-core/src/lib/credentials/credentials.js

```diff
@@ -267,6 +287,12 @@ const Credentials = SparkPlugin.extend({
           this.supertoken = attrs.authorization;
         }
       }
+
+      // schedule refresh
+      if (this.supertoken) {
+        const timeoutLength = this._calcRefreshTimeout(this.supertoken.expires_in);
```

Comment:
There's an edge case this doesn't cover. when you load a token from storage, expires_in doesn't mean anything any more (since it's a duration from the time the token was received).